packages/wonder-blocks-core/util/enumerate-scroll-ancestors.test.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {mount} from "enzyme";
+
+import enumerateScrollAncestors from "./enumerate-scroll-ancestors.js";
+
+describe("enumerateScrollAncestors", () => {
+    test("@@iterator() method returns iterator", () => {
+        // Arrange
+        const enumerator = enumerateScrollAncestors((null: any));
+
+        // Act
+        // Flow doesn't want us to index  $FlowFixMe
+        const result = enumerator[Symbol.iterator]();
+
+        // Assert
+        expect(result).not.toBeUndefined();
+        expect(result).toBe(result[Symbol.iterator]());
+    });
+
+    test("if passed an invalid parameter, yields nothing", () => {
+        // Arrange
+        const param = (("this is not an element": any): Element);
+
+        // Act
+        const result = Array.from(enumerateScrollAncestors(param));
+
+        // Assert
+        expect(result.length).toBe(0);
+    });
+
+    test("if there are no scroll parents, yields documentElement", (done) => {
+        const arrange = (actAndAssert) => {
+            // Arrange
+            const nodes = (
+                <div>
+                    <div ref={actAndAssert}>Test</div>
+                </div>
+            );
+
+            mount(nodes);
+        };
+
+        const actAndAssert = (ref) => {
+            if (ref) {
+                const domNode = ((ReactDOM.findDOMNode(ref): any): Element);
```

Comment:
nit: the `ref` could be typed as a `Node`.

Reply:
I don't think that's right. If it is specified as a `Node`, `findDOMNode` gets quite upset if I do that. Since `flow` isn't complaining without the type, I am inclined to leave it untyped.

Reply:
interesting... I was sure that would work, I guess I have to update my understanding of how ref callbacks work.

Reply:
I believe the type for `findDOMNode` has to be `?(Element | React.Component<*>)`.